crates/rslint_parser/src/syntax/typescript.rs

```diff
@@ -982,7 +986,8 @@ pub fn ts_non_array_type(p: &mut Parser) -> Option<CompletedMarker> {
 			}
 		}
 		JS_NUMBER_LITERAL | JS_STRING_LITERAL | TRUE_KW | FALSE_KW | JS_REGEX_LITERAL => Some(
-			literal_expression(p)
+			parse_literal_expression(p)
+				.ok()
 				.unwrap()
 				.precede(p)
```

Comment:
Nit: You can use `parse_literal_expression(p).precede_or_missing()`
```suggestion
			parse_literal_expression(p)
				.precede_or_missing(p)
```